proto/src/main/java/com/google/gson/protobuf/ProtoTypeAdapter.java

```diff
@@ -174,13 +176,39 @@ public Builder addSerializedEnumValueExtension(
       return this;
     }
 
+    /**
+     * Sets or unsets a flag that, when set, causes the adapter to use the `json_name` field option
+     * from a proto field for serialization. This `json_name` option is an annotation applied to
+     * proto fields that cannot be read from the descriptor's options
+     * (`FieldDescriptor::getOptions`) as it is only available via `FieldDescriptor::getJsonName`.
```

Comment:
Here I think you could write `{@link FieldDescriptor#getOptions}` and `{@link FieldDescriptor#getJsonName}` since we do import `FieldDescriptor`.

Reply:
Is this sentence about `getOptions` and `getJsonName` actually relevant for users of this adapter, or is this an implementation detail and we should omit it?

Reply:
i meant it as a way to explain why users of the adapter expect to see this method here instead of just using the `addSerializedNameExtension` which might be their initial intuition. i'm cool with removing it if you think it doesn't help.